boost/math/tools/config.hpp

```diff
@@ -10,46 +11,135 @@
+#endif
+#else
+#define BOOST_IF_CONSTEXPR if
+#define BOOST_NO_CXX17_IF_CONSTEXPR
+#define BOOST_NO_CXX17_HDR_EXECUTION
+#endif
+
+#if __cpp_lib_gcd_lcm >= 201606L
+#define BOOST_MATH_HAS_CXX17_NUMERIC
+#endif
+
+#define BOOST_JOIN(X, Y) BOOST_DO_JOIN(X, Y)
+#define BOOST_DO_JOIN(X, Y) BOOST_DO_JOIN2(X,Y)
+#define BOOST_DO_JOIN2(X, Y) X##Y
+
+#define BOOST_STRINGIZE(X) BOOST_DO_STRINGIZE(X)
+#define BOOST_DO_STRINGIZE(X) #X
+
+#ifdef BOOST_DISABLE_THREADS // No threads, do nothing
+// Detect thread support via STL implementation
+#elif defined(__has_include)
+#  if !__has_include(<thread>) || !__has_include(<mutex>) || !__has_include(<future>) || !__has_include(<atomic>)
+#     define BOOST_DISABLE_THREADS
+#  else
+#     define BOOST_HAS_THREADS
+#  endif 
+#else
+#  define BOOST_HAS_THREADS // The default assumption is that the machine has threads
+#endif // Thread Support
+
+#ifdef BOOST_DISABLE_THREADS
+#  define BOOST_NO_CXX11_HDR_ATOMIC
+#  define BOOST_NO_CXX11_HDR_FUTURE
+#  define BOOST_NO_CXX11_HDR_THREAD
+#  define BOOST_NO_CXX11_THREAD_LOCAL
+#endif // BOOST_DISABLE_THREADS
+
+#ifdef __GNUC__
+#  if !defined(__EXCEPTIONS) && !defined(BOOST_NO_EXCEPTIONS)
+#     define BOOST_NO_EXCEPTIONS
+#  endif
+   //
+   // Make sure we have some std lib headers included so we can detect __GXX_RTTI:
+   //
+#  include <algorithm>  // for min and max
+#  include <limits>
+#  ifndef __GXX_RTTI
+#     ifndef BOOST_NO_TYPEID
+#        define BOOST_NO_TYPEID
+#     endif
+#     ifndef BOOST_NO_RTTI
+#        define BOOST_NO_RTTI
+#     endif
+#  endif
+#endif
+
+#endif // BOOST_MATH_STANDALONE
+
+// Support compilers with P0024R2 implemented without linking TBB
+// https://en.cppreference.com/w/cpp/compiler_support
+#if !defined(BOOST_NO_CXX17_HDR_EXECUTION) && defined(BOOST_HAS_THREADS)
+#  define BOOST_MATH_EXEC_COMPATIBLE
+#endif
+
 #include <algorithm>  // for min and max
-#include <boost/config/no_tr1/cmath.hpp>
+#include <limits>
+#include <cmath>
 #include <climits>
 #include <cfloat>
 #if (defined(macintosh) || defined(__APPLE__) || defined(__APPLE_CC__))
 #  include <math.h>
 #endif
-#ifndef BOOST_NO_LIMITS_COMPILE_TIME_CONSTANTS
-#  include <limits>
-#endif
 
 #include <boost/math/tools/user.hpp>
 
 #if (defined(__NetBSD__) || defined(__EMSCRIPTEN__)\
    || (defined(__hppa) && !defined(__OpenBSD__)) || (defined(__NO_LONG_DOUBLE_MATH) && (DBL_MANT_DIG != LDBL_MANT_DIG))) \
    && !defined(BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS)
-#  define BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
-#endif
-#if BOOST_WORKAROUND(BOOST_BORLANDC, BOOST_TESTED_AT(0x582))
-//
-// Borland post 5.8.2 uses Dinkumware's std C lib which
-// doesn't have true long double precision.  Earlier
-// versions are problematic too:
-//
-#  define BOOST_MATH_NO_REAL_CONCEPT_TESTS
-#  define BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
-#  define BOOST_MATH_CONTROL_FP _control87(MCW_EM,MCW_EM)
-#  include <float.h>
+//#  define BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
```

Comment:
@mkoeppe This is the only change I'm currently worried about -- do we need a new patch for this or do things work as expected in vanilla Boost 1.80 and we can remove it altogether?

Reply:
Upstream's new version looks fine, so I think you can remove the patch.